src/index.js

```diff
@@ -0,0 +1,56 @@
+import { pipe, pipeCompatibleMap } from './utility.js';
+import { wordWrap } from './accessibility.js';
+import { formatLine } from './format.js';
+
+//item  layer
+const isSeparator = (item) => item === separator;
+
+const hasSubmenu = (item) => !!item.submenu;
+
+const getSubmenu = (item) =>
+  item.submenu.map((e) => ({
+    ...e,
+    submenuLevel: item.submenuLevel ? item.submenuLevel + 1 : 1,
+  }));
+
+// deno-lint-ignore no-unused-vars
+const ignoreSubmenu = ({ submenu, ...item }) => item;
+
+const formatItem = pipe(wordWrap, pipeCompatibleMap(formatLine));
+
+//array layer
+const addItem = (arr, item) => [
+  ...arr,
+  ...formatItem(item),
+];
+
+const addItemWithSubmenu = (arr, item) => [
+  ...addItem(arr, ignoreSubmenu(item)),
+  ...pipe(getSubmenu, xbar)(item),
+];
+
+const addSeparator = (arr) => [
+  ...arr,
+  formatLine({ text: '---' }),
+];
+
+//entry layer
+/**
+ * Prints output to xbar and returns printed output as an array of strings
+ * @param layout - array of menu items
+ * @returns {string[]} array of strings that were printed as menu items
+ */
+export const xbar = (layout) =>
+  layout.reduce((menuItems, item) => {
+    switch (true) {
+      case (isSeparator(item)):
```

Comment:
maybe `needsSeparator`. I'm not sure why you would add a separator to a separator

Reply:
How does `includeSeparator` seem? The idea was when the function runs into a separator it adds it to the layout but you are correct that right now it does read like an additional separator is getting added